File: m/rls/rlb/strep.c
/* file -  strep - replace substring */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_rep - replace string if found */
char *st_rep(
char *mod ,
char *rep ,
char *str )
{ char *pnt ;
  if( (pnt = st_fnd (mod, str)) == NULL)return ( NULL );
  if( rep == NULL)return ( pnt );
  return ( (st_exc (rep, pnt, st_len (mod))));
} 

File: m/rls/strep.r
file	strep - replace substring
include	rid:rider
include	rid:stdef

code	st_rep - replace string if found

  func	st_rep
	mod : * char			; model string
	rep : * char			; replacement string
	str : * char			; edit string
	()  : * char			; past replacement or <>
					; at replacement if rep eq <>
  is	pnt : * char			; pointer
	pnt = st_fnd (mod, str)		; locate the model
	pass <>				; not found
	reply pnt if rep eq <>		; just want to find it
	st_exc (rep, pnt, st_len (mod))	; replace it
	reply that			; past replacement
  end
